supabase/migrations/20240425205531_create_auth_challege_table.sql

```diff
@@ -0,0 +1,91 @@
+    "user_id" uuid not null references auth.users (id) on delete cascade,
+    -- Ensure the challenge is a hex encoded 256-bit hash (sha256)
+    "challenge" citext not null constraint auth_challenges_challenge_check check (
+        (length(("challenge")::text) = 64)
+        and ("challenge" ~ '^[0-9a-fA-F]{64}$'::citext)
+    ),
+    -- Define the creation time and expiration time for the challenge aka:
+    -- its validity window in which is can be signed and verified before another
+    -- challenge has to be made.
+    "created_at" timestamp with time zone not null default current_timestamp,
+    "expires_at" timestamp with time zone not null default current_timestamp + interval '15 minute'
+);
+
+-- Create a unique primary key
+create unique index "auth_challenges_pkey" on "public"."auth_challenges" using btree ("id");
+
+-- Enable row level security, ensuring only the service role can interact with
+-- this table, aka our supanasrAdmin helper type. This prevents any spoofing
+-- attacks and disdables users from seeing who is recovering their account
+-- which could lead to them devising attack vectors to gain access to the accounts
+alter table "public"."auth_challenges" enable row level security;
+
+-- Create a copy of the users table only supabaseAdmin can view for phone
+-- number usage in challenges
+create view "public"."users" as select * from "auth"."users";
+revoke all on "public"."users" from anon, authenticated, public;
+grant all on "public"."users" to service_role;
+
+
+-- Secure the table granting all permissions to the service role
+revoke all on "public"."auth_challenges" from anon, authenticated, public;
+grant all on table "public"."auth_challenges" to service_role;
+
+-- Add the primary key using the auth_challenges table's random id column
+alter table "public"."auth_challenges" add constraint "auth_challenges_pkey"
+primary key using index "auth_challenges_pkey";
+
+-- Ensure the user_id field is unique and only one challenge exists per user row
+-- index the user_id field using a `btree` for fast retrieval during verification
+create unique index "auth_challenges_user_id_idx" on "public"."auth_challenges"
+using btree ("user_id");
+
+-- Add the r`"user_id"r` field is a unique key which will fail an insert operation
+-- if it already exists in the table. This will allow us to update on conflict
+-- or the `user_id` in question with the the new values rejected by the insert
+-- operation  Minus "user_id", the rest are then passed to a following update 
+-- function which will update the `created_at` and `expires_at` as well as
+-- the `challenge` field if the `user_id` supplied along with a new derived
+-- "id" field for the  challenge row in the table.
+alter table "public"."auth_challenges" add constraint "auth_challenges_user_id_idx"
+unique using index "auth_challenges_user_id_idx";
+
+-- Create a function to update the table and add a challenge
+create or replace function upsert_auth_challenges(
+    userid uuid,
```

Comment:
```suggestion
    user_id uuid,
```

I prefer to use `#variable_conflict use_column` and then set an alias in the `declare` section.

```sql
declare
            _user_id alias for $1 
            _auth_id uuid;
            _created timestamptz;
            _expires timestamptz;
            _new_challenge auth_challenges;
    begin
```

Reply:
https://github.com/0xsend/sendapp/blob/41d76df1b91b03b6d6181b7c6ec5a8882c6fea7c/supabase/migrations/20240426211951_create_send_accounts_add_webauthn_credential.sql#L21

Reply:
changes made. please review this [commit](https://github.com/0xsend/sendapp/pull/437/commits/06edbac53133bd42e148d7343fc8458eec412d37)